File: src/mgos_arduino_liquidcrystal_i2c.cpp
#include "mgos_arduino_liquidcrystal_i2c.h"

LiquidCrystal_I2C* mgos_liquidcrystal_i2c_create(int lcd_Addr, int lcd_cols, int lcd_rows) {
  return new LiquidCrystal_I2C(lcd_Addr,lcd_cols,lcd_rows);
}

void mgos_liquidcrystal_i2c_close(LiquidCrystal_I2C* lcd) {
  if (lcd != nullptr) {
    delete lcd;
    lcd = nullptr;
  }
}

void mgos_liquidcrystal_i2c_init(LiquidCrystal_I2C* lcd){
  if (lcd == nullptr) return;
  return lcd->init();
}

void mgos_liquidcrystal_i2c_backlight(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->backlight();
}

void mgos_liquidcrystal_i2c_noBacklight(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->noBacklight();
}

int mgos_liquidcrystal_i2c_getAddr(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return 0;
  return lcd->getAddr();
}

void mgos_liquidcrystal_i2c_setCursor(LiquidCrystal_I2C* lcd, int col, int row){
  if (lcd == nullptr) return;
  return lcd->setCursor(col, row);
}

int mgos_liquidcrystal_i2c_print(LiquidCrystal_I2C* lcd, const char *str) {
  if (lcd == nullptr) return 0;
  return lcd->print(str);
}

int mgos_liquidcrystal_i2c_printChar(LiquidCrystal_I2C* lcd, const char *c) {
  if (lcd == nullptr) return 0;
  return lcd->print(c);
}

void mgos_liquidcrystal_i2c_clear(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->clear();
}

void mgos_liquidcrystal_i2c_home(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->home();
}

void mgos_liquidcrystal_i2c_noDisplay(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->noDisplay();
}

void mgos_liquidcrystal_i2c_display(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->display();
}

void mgos_liquidcrystal_i2c_noBlink(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->noBlink();
}

void mgos_liquidcrystal_i2c_blink(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->blink();
}

void mgos_liquidcrystal_i2c_noCursor(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->noCursor();
}

void mgos_liquidcrystal_i2c_cursor(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->cursor();
}

void mgos_liquidcrystal_i2c_scrollDisplayLeft(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->scrollDisplayLeft();
}

void mgos_liquidcrystal_i2c_scrollDisplayRight(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->scrollDisplayRight();
}

void mgos_liquidcrystal_i2c_printLeft(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->printLeft();
}

void mgos_liquidcrystal_i2c_printRight(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->printRight();
}

void mgos_liquidcrystal_i2c_leftToRight(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->leftToRight();
}

void mgos_liquidcrystal_i2c_rightToLeft(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->rightToLeft();
}

void mgos_liquidcrystal_i2c_shiftIncrement(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->shiftIncrement();
}

void mgos_liquidcrystal_i2c_shiftDecrement(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->shiftDecrement();
}

void mgos_liquidcrystal_i2c_autoscroll(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->autoscroll();
}


void mgos_liquidcrystal_i2c_noAutoscroll(LiquidCrystal_I2C* lcd) {
  if (lcd == nullptr) return;
  return lcd->noAutoscroll();
}

// void mgos_liquidcrystal_i2c_createChar(LiquidCrystal_I2C* lcd, int location, int *charmap) {
//   if (lcd == nullptr) return;
//   return lcd->createChar(location, charmap);
// }

int mgos_liquidcrystal_i2c_write(LiquidCrystal_I2C* lcd, int value) {
  if (lcd == nullptr) return 0;
  return lcd->write(value);
}

void mgos_liquidcrystal_i2c_command(LiquidCrystal_I2C* lcd, int value) {
  if (lcd == nullptr) return;
  return lcd->command(value);
}

File: mjs_fs/mgos_arduino_liquidcrystal_i2c.js


let LiquidCrystalI2C = {
  _create: ffi('void *mgos_liquidcrystal_i2c_create(int, int, int)'),
  _close: ffi('void mgos_liquidcrystal_i2c_close(void *)'),
  _init: ffi('void mgos_liquidcrystal_i2c_init(void *)'),
  _backlight: ffi('void mgos_liquidcrystal_i2c_backlight(void *)'),
  _noBacklight: ffi('void mgos_liquidcrystal_i2c_noBacklight(void *)'),
  _getAddr: ffi('int mgos_liquidcrystal_i2c_getAddr(void *)'),
  _setCursor: ffi('void mgos_liquidcrystal_i2c_setCursor(void *, int, int)'),
  _print: ffi('size_t mgos_liquidcrystal_i2c_print(void *, char *)'),
  _printChar: ffi('size_t mgos_liquidcrystal_i2c_printChar(void *, char *)'),
  _clear: ffi('void mgos_liquidcrystal_i2c_clear(void *)'),
  _home: ffi('void mgos_liquidcrystal_i2c_home(void *)'),
  _noDisplay: ffi('void mgos_liquidcrystal_i2c_noDisplay(void *)'),
  _display: ffi('void mgos_liquidcrystal_i2c_display(void *)'),
  _noBlink: ffi('void mgos_liquidcrystal_i2c_noBlink(void *)'),
  _blink: ffi('void mgos_liquidcrystal_i2c_blink(void *)'),
  _noCursor: ffi('void mgos_liquidcrystal_i2c_noCursor(void *)'),
  _cursor: ffi('void mgos_liquidcrystal_i2c_cursor(void *)'),
  _scrollDisplayLeft: ffi('void mgos_liquidcrystal_i2c_scrollDisplayLeft(void *)'),
  _scrollDisplayRight: ffi('void mgos_liquidcrystal_i2c_scrollDisplayRight(void *)'),
  _printLeft: ffi('void mgos_liquidcrystal_i2c_printLeft(void *)'),
  _printRight: ffi('void mgos_liquidcrystal_i2c_printRight(void *)'),
  _leftToRight: ffi('void mgos_liquidcrystal_i2c_leftToRight(void *)'),
  _rightToLeft: ffi('void mgos_liquidcrystal_i2c_rightToLeft(void *)'),
  _shiftIncrement: ffi('void mgos_liquidcrystal_i2c_shiftIncrement(void *)'),
  _shiftDecrement: ffi('void mgos_liquidcrystal_i2c_shiftDecrement(void *)'),
  _noBacklight: ffi('void mgos_liquidcrystal_i2c_noBacklight(void *)'),
  _autoscroll: ffi('void mgos_liquidcrystal_i2c_autoscroll(void *)'),
  _noAutoscroll: ffi('void mgos_liquidcrystal_i2c_noAutoscroll(void *)'),
  // _createChar: ffi('void mgos_liquidcrystal_i2c_createChar(void *, int, int[])'),
  _write: ffi('size_t mgos_liquidcrystal_i2c_write(void *, int)'),
  _command: ffi('void mgos_liquidcrystal_i2c_command(void *, int)'),

  create: function(lcd_Addr, lcd_cols, lcd_rows) {
    let obj = Object.create(LiquidCrystalI2C._proto);
    // Initialize DallasTemperature library.
    // Return value: handle opaque pointer.
    obj.lcd = LiquidCrystalI2C._create(lcd_Addr, lcd_cols, lcd_rows);
    return obj;
  },

  _proto: {
    close: function() {
      return LiquidCrystalI2C._close(this.lcd);
    },

    init: function() {
      return LiquidCrystalI2C._init(this.lcd);
    },

    backlight: function() {
      return LiquidCrystalI2C._backlight(this.lcd);
    },

    noBacklight: function() {
      return LiquidCrystalI2C._noBacklight(this.lcd);
    },
  }
}
